src/engine/shared/network.cpp

```diff
@@ -39,29 +39,43 @@ void CNetRecvUnpacker::Start(const NETADDR *pAddr, CNetConnection *pConnection,
 // TODO: rename this function
 int CNetRecvUnpacker::FetchChunk(CNetChunk *pChunk)
 {
-	CNetChunkHeader Header;
-	unsigned char *pEnd = m_Data.m_aChunkData + m_Data.m_DataSize;
+	if(!m_Valid)
+	{
+		// Receiver not initialized
+		return 0;
+	}
+	else if(m_Data.m_DataSize < 0)
+	{
+		// Packet decompression failed
+		Clear();
+		return 0;
+	}
+
+	const unsigned char *const pEnd = m_Data.m_aChunkData + m_Data.m_DataSize;
 
 	while(true)
 	{
-		unsigned char *pData = m_Data.m_aChunkData;
-
-		// check for old data to unpack
-		if(!m_Valid || m_CurrentChunk >= m_Data.m_NumChunks)
+		if(m_CurrentChunk >= m_Data.m_NumChunks)
 		{
+			// No more chunks to unpack from this packet
 			Clear();
 			return 0;
 		}
 
+		unsigned char *pData = m_Data.m_aChunkData;
+
 		// TODO: add checking here so we don't read too far
+		const int HeaderSplit = m_pConnection->m_Sixup ? 6 : 4;
```

Comment:
```suggestion
		const int HeaderLen = m_pConnection->m_Sixup ? 6 : 4;
```
Is this just the header length?